### PR TITLE
Add a no_overlay option for the rkt task config.

### PR DESCRIPTION
**Pull request by kmalec**
Directly relates to https://github.com/rkt/rkt/issues/1922 - for various reasons, I have to keep up with RHEL 7 systems, and --no-overlay for rkt run seems to help with the bug above.

### CODE REVIEW

File: website/source/docs/drivers/rkt.html.md [hunk 1]
```diff
@@ -88,6 +88,9 @@ The `rkt` driver supports the following configuration in the job spec:
 
 * `debug` - (Optional) Enable rkt command debug option.
 
+* `no_overlay` - (Optional) When enabled, will use --no-overlay=true flag for 'rkt run'.
```
Comment (schmichael) [inline website/source/docs/drivers/rkt.html.md]: Backticks around the command line flag:
```
... use `--no-overlay=true` flag for ...
```

**Comment by kmalec** [comment]
The build I actually have running, but checking some test jobs from yesterday, I see a few `alloc/UUID/task-name/secrets` leftover tmpfs mounts in one of the systems. 
rkt gc runs fine now, so `--no-overlay` hopefully helps, but I'm not sure where those `secrets` mounts come from. My environment is a bit volatile right now, maybe unrelated to this change.

**Comment by schmichael** [comment]
Fixed conflicts and am waiting on Travis.

**Comment by kmalec** [comment]
Okay, I did some more basic tests with a simple centos7 ACI on RHEL7.3 host.
* the no_overlay option resolves the GC issue referenced in the commit - for the price of performance hit of not using overlayfs, I can now `rkt gc` without those `Device or resource busy` on stage2 rootfs
* the leaking mountpoints of Nomad (`tmpfs          tmpfs     1.0M     0  1.0M   0% /data/nomad-data/alloc/e81f7fec-0d56-5970-6af2-3019343201b9/linuk-task/secrets`) don't seem to be caused by this commit, as they happen with 0.5.6 client agent too. Not sure if that's tracked yet.

Was just worried this PR could be the cause of those mountpoints, but no :)

**Comment by schmichael** [comment]
Closing in favor of #2702 -- I had to fix a merge conflict and needed to push up the changes. Sorry! Will get #2702 merged as soon as Travis passes.

**Pull request closed without merging**